examples/cpp/main.cpp

```diff
@@ -231,6 +231,56 @@ void TestThirdParty() {
       false, "image");
 }
 
+void TestClassId() {
+  Engine engine(
+      "###element\n"
+      "##.ads\n"
+      "##.element\n"
+      "###ads > #element\n"
+      "##a[href^=\"test.com\"]\n"
+      "###block\n"
+      "###block + .child\n"
+  );
+  std::vector<std::string> classes = std::vector<std::string>();
+  std::vector<std::string> ids = std::vector<std::string>();
+  std::vector<std::string> exceptions = std::vector<std::string>();
+  std::string stylesheet = engine.classIdStylesheet(classes, ids, exceptions);
+  assert(stylesheet == "");
+
+  classes = std::vector<std::string>({"ads", "no-ads"});
+  ids = std::vector<std::string>({"element"});
+  exceptions = std::vector<std::string>();
+  stylesheet = engine.classIdStylesheet(classes, ids, exceptions);
+  assert(stylesheet == ".ads,#element{display:none !important;}");
+
+  classes = std::vector<std::string>({"element", "a"});
+  ids = std::vector<std::string>({"block", "ads", "a"});
+  exceptions = std::vector<std::string>({"#block"});
+  stylesheet = engine.classIdStylesheet(classes, ids, exceptions);
+  assert(stylesheet == ".element,#block + .child,#ads > #element{display:none !important;}");
+}
+
+void TestHostnameCosmetics() {
```

Comment:
@antonok-edm could you also add a test with a subdomain to show what the expected usage is w.r.t. doing eTLD+1 or not.